### PR TITLE
Create interactive boxes to enable clearer user journeys

### PR DESCRIPTION
**Pull request by anjleeg-gcai**
Still in progress, todo the following:
- [x] Add additional description to each of the drop downs to describe what is contained in each of the notebooks
- [x] Clear up some of the text surronding
- [x] Change title for dropdown section (Not keen on notebook garden - open to suggestions)
- [x] Double check notebooks are relevant for each section
- [x] Play with formatting in Paperspace

### CODE REVIEW

**Comment by anjleeg-gcai** [comment]
Screenshot 2023-02-16 at 16 04 53

It looks like this on the jupyter notebooks, this is just a draft currently.

File: README_first.ipynb [hunk 1]
```diff
@@ -1,60 +1,109 @@
 {
```
Comment (payoto) [inline README_first.ipynb:1]: <p>Wouldn't call it advanced would call it, "Useful tips" or "good to know", maybe not nested. The reason for my thought process here is that we need the users to know this exists so they can solve their problems.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/Graphcore-HuggingFace-fork/pull/22/discussion/#comment-1108743990'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='5'/>

**Comment by payoto** [comment]
Also i like headings which havee active verbs:

"Use 🤗 model for inference"
"Fine-tune an off the shelf model on your dataset"
"Build your own model"

File: README_first.ipynb [hunk 2]
```diff
@@ -1,60 +1,124 @@
...
```
Comment (payoto) [inline README_first.ipynb:1]: <p>I would:</p><p> </p><ul><li>drop the exclamation mark(s)</li><li>Split with two sub-heading for the domain of the task (audio and NLP)</li><li>Have basically the relevant tabular entries from the Hugging Face optimum notebooks/README.md</li></ul><p> </p><p>Also double check language:</p><ul><li>"our our"</li><li>"to help you get you started"</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/Graphcore-HuggingFace-fork/pull/22/discussion/#comment-1112887520'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

Comment (payoto) [inline README_first.ipynb:1]: <p>Ideas are the wrong way round, should be:</p><ul><li>2 Lines of code</li><li>NLP: Sentiment Analysis and NER</li><li>Wav2Vec2 inference</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/Graphcore-HuggingFace-fork/pull/22/discussion/#comment-1112887524'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

Comment (payoto) [inline README_first.ipynb:1]: <p>I don't think the emphasis should be on porting, what they want to do is use their model, porting is the unfortunate reality.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/Graphcore-HuggingFace-fork/pull/22/discussion/#comment-1112887527'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

Comment (anjleeg-gcai) [reply README_first.ipynb:1]: <p>Do you have a suggestion for a different header title?</p><p> </p>

File: README_first.ipynb [hunk 3]
```diff
@@ -1,60 +1,160 @@
...
```
Comment (payoto) [inline README_first.ipynb:1]: <p>I think you can remove the first line: "Here you can learn how to use Hugging Face checkpoints directly for inference."</p><p>I would put this box above the finetuning.</p><blockquote>Our NLP notebooks show how to run inference with only 2 lines of code!</blockquote><p>I think 2 is a bit optimistic :)</p><p> </p><p>Maybe rephrase this box to something like:</p><p> </p><p>See how to solve NLP tasks on the IPU in 5 lines of code, and turn it into an app in 10 lines in our NLP notebooks:</p><p> </p><pre class="ql-syntax" spellcheck="false">    natural-language-processing/ 
        └───sentiment_analysis.ipynb 
        └───name-entity-extraction.ipynb 
</pre><p>Use the sentiment-analysis pipeline to quickly evaluate pre-trained models and check out how to use Gradio with the IPU to prototype a web application doing fast token classification in our name entity extraction notebook.</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/Graphcore-HuggingFace-fork/pull/22/discussion/#comment-1117050680'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

Comment (payoto) [inline README_first.ipynb:1]: <p>"an external that" missing the word model?</p><p> </p><p>also "external" is a bad word for describing a model, external to what? Maybe "and learn how to train a model which is not yet supported by <code>optimum-graphcore</code>"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/Graphcore-HuggingFace-fork/pull/22/discussion/#comment-1117050683'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

Comment (payoto) [inline README_first.ipynb:1]: <p>Suggestion for title: Image generation with Stable Diffusion</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/Graphcore-HuggingFace-fork/pull/22/discussion/#comment-1117050688'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='4'/>

**Comment by payoto** [comment]
I find the formatting really hard to parse, can you try:

| file     | Content |
| ----------- | ----------- |
| audio-processing/wav2vec2-fine-tuning-checkpoint.ipynb      | Check out how to fine-tune a pre-trained wav2vec 2.0 model with PyTorch on the IPU.       |
| image-classification/image-classification.ipynb   | > notebook will show you how to preprocess the data and fine-tune a pretrained model on image classification.        |

Maybe the other way round with content and file?

**Comment by anjleeg-gcai** [comment]
For all of the examples or just for those two? If I'm being honest, I'm not a huge fan of the way the tables look.

<details><summary><big> Fine-tune our models on your datasets</summary><br/>

Here are our notebooks on how to use datasets to fine-tune HuggingFace checkpoints.
    
| Content | Location |
| :-- | :-- |
| Check out how to fine-tune a pre-trained wav2vec 2.0 model with PyTorch on the IPU.| audio-processing/wav2vec2-fine-tuning-checkpoint.ipynb |
| This notebook will show you how to preprocess the data and fine-tune a pretrained model on image classification.| image-classification/image-classification.ipynb  

#### Natural Language Processing

Our NLP notebooks will teach you how to fine-tuning your models for a range of different tasks. Show how to preprocess the data and fine-tune a pretrained model on a range of datasets including WMT, SWAG, SQUAD and many more in our NLP notebooks:

```
    natural-language-processing/
        └───translation.ipynb
        └───other-use-cases/
            └─── multiple_choice.ipynb
            └─── question_answering.ipynb
            └─── text_classification.ipynb
            └─── token_classification.ipynb
```

</details>

File: README_first.ipynb [hunk 4]
```diff
@@ -1,60 +1,342 @@
...
+    "<tr>\n",
+    "<td align=\"left\">\n",
+    "    <td> <img align=\"left\" width=\"18\" src=\"./images/folder_logo.png\" /> useful-tips/ </td>\n",
+    "    <td> </td>\n",
+    "</td>\n",
+    "</tr>\n",
+    "<tr>\n",
+    "<td align=\"left\">\n",
+    "    <td> &nbsp&nbsp&nbsp <img align=\"left\" width=\"18\" src=\"./images/jupyter_logo.png\" /> managing_ipu_resources.ipynb </td>\n",
+    "    <td width = 70%> See how to monitor IPU usage, releasing IPUs when you are not using them, and then how to re-attach your model to the IPU to use it again. \n",
```
Comment (alex-titterton) [inline README_first.ipynb]: "... releasing IPUs..." --> "... _how to release_ IPUs..."?

File: README_first.ipynb [hunk 5]
```diff
@@ -1,60 +1,342 @@
...
+    "<tr>\n",
+    "<td align=\"left\">\n",
+    "    <td> &nbsp&nbsp&nbsp <img align=\"left\" width=\"18\" src=\"./images/jupyter_logo.png\" /> sentiment_analysis.ipynb </td>\n",
+    "    <td> Use the sentiment-analysis pipeline to quickly evaluate pre-trained models on the IPU.  </td>\n",
+    "</td>\n",
+    "</tr>\n",
+    "<tr>\n",
+    "<td align=\"left\">\n",
+    "    <td> &nbsp&nbsp&nbsp <img align=\"left\" width=\"18\" src=\"./images/jupyter_logo.png\" /> name-entity-extraction.ipynb </td>\n",
+    "    <td> Use Gradio and pipelines to prototype a web application doing fast token classification.  </td>\n",
```
Comment (alex-titterton) [inline README_first.ipynb]: This might be fine, and I might just not be familiar, but "Use Gradio and pipelines..." -- is this correct, or is something missing before "pipelines"?

File: README_first.ipynb [hunk 6]
```diff
@@ -1,60 +1,342 @@
...
+    "</tr>\n",
     "\n",
-    "These worked examples will help you get started running your application on the IPU. They also demonstrate techniques such as data-parallelism and pipelining to increase performance by using multiple IPUs.\n",
+    "</table>\n",
     "\n",
-    "## Getting started\n",
+    "<br><br><br><br>\n",
     "\n",
-    "While no specific user setup is required to train IPU models on Gradient, many of the notebooks in this runtime will download from and push model checkpoints to the Hugging Face model Hub. For the best experience we recommend that you generate a [User Access Token](https://huggingface.co/docs/hub/security-tokens) with \"Write\" access to your Hugging Face account. By doing so, the checkpoints from the models you fine tune will automatically be saved and be available for you to use later. Without this token, some models may fail to train and you may need to set the `push_to_hub` argument to false in the `IPUTrainingArguments` class before starting IPU training.\n",
+    "For dedicated finetuning, inference, advanced ML and IPU related notebooks checkout the following notebooks:\n"
```
Comment (alex-titterton) [inline README_first.ipynb]: "checkout" or "check out"?

**Review by alex-titterton — COMMENTED** [review]
In-line comments added

File: README_first.ipynb (hunk 5 above)
Comment (anjleeg-gcai) [reply README_first.ipynb]: I've added "HuggingFace" pipelines to make it more clear

**Review by payoto — APPROVED** [review]
Looks totally amazing!

**Review by evawoodbridge — APPROVED** [review]
LGTM
